File: android/app/src/main/java/org/fs/news/model/SourceModel.kt
/*
 *  Copyright (C) 2019 Fatih, News Stand Android Kotlin.
 *  
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.fs.news.model

import org.fs.architecture.mvi.common.Model
import org.fs.architecture.mvi.common.SyncState
import org.fs.news.model.entity.Source

data class SourceModel(override val state: SyncState, override val data: List<Source>): Model<List<Source>>(state, data)

File: ios/NewsStand/Model/SourceModel.swift
//
//  SourceModel.swift
//  NewsStand
//
//  Created by Fatih Şen on 29.06.2019.
//  Copyright © 2019 Fatih Şen. All rights reserved.
//

import Foundation
import MVICocoa

public struct SourceModel: Model {

  public static let empty = SourceModel(state: .idle, data: [])

  public var state: SyncState
  public var data: [Source]

  public func copy(state: SyncState? = nil, data: [Source]? = nil) -> SourceModel {
    return SourceModel(state: state ?? self.state, data: data ?? self.data)
  }
}
